src/ucp/core/ucp_ep.c

```diff
@@ -375,7 +375,8 @@ ucs_status_t ucp_ep_init_create_wireup(ucp_ep_h ep, unsigned ep_init_flags,
     /* all operations will use the first lane, which is a stub endpoint before
      * reconfiguration */
     key.am_lane             = 0;
-    if (ucp_worker_sockaddr_is_cm_proto(ep->worker)) {
+    if (ucp_worker_sockaddr_is_cm_proto(ep->worker) &&
+        ucp_ep_init_flags_has_cm(ep_init_flags)) {
```

Comment:
in which flow ucp_ep_init_flags_has_cm would be 0 here? MPI?

Reply:
in case when ep is connected to worker address, yes, IMPI for example